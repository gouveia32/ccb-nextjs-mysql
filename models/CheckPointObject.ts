export interface CheckPointInterface {
    id?: number | string;
    text: string;
    checked: boolean;
  }
  
  export type CheckPointType = CheckPointInterface;
  
  export const CheckPointObject: CheckPointType = {
           id: 0,
           text: "",
           checked: false,
         };
  
  export enum cCheckPoint {
    id = 'id',
    text = 'text',
    checked = 'checked',
  }