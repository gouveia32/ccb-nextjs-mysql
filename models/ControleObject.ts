export interface ControleInterface {
  id?: number | string;
  texto: string;
  marcado: boolean;
}

export type TipoControle = ControleInterface;

export const ControleObject: TipoControle = {
         id: 0,
         texto: "",
         marcado: false,
       };

export enum cControle {
  id = 'id',
  texto = 'texto',
  marcado = 'marcado',
}
