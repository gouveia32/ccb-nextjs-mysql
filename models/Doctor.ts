export interface DoctorModel {
  id: string;
  name: string;
  email: string;
  image?: string;
}

export type DoctorType = DoctorModel;

export const cDoctorModel = {
  id: "id",
  name: "name",
  email: "email",
  image: '',
};

export const DoctorObject: DoctorType = {
  id: "",
  name: "",
  email: "",
  image: '',
};

export function isDoctorType(object: any): object is DoctorType {
  return "name" in object;
}
