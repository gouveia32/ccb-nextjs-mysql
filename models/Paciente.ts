export interface PacienteModel {
  id: string;
  nome: string;
}

export type TipoPaciente = PacienteModel;

export const cPacienteModel = {
  id: "id",
  nome: "nome"
};

export const PacienteObject: TipoPaciente = {
  id: "",
  nome: "",
};

export function isPacienteType(object: any): object is TipoPaciente {
  return "nome" in object;
}
