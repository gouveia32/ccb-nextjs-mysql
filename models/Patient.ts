export interface PatientModel {
  id: string;
  name: string;
}

export type PatientType = PatientModel;

export const cPatientModel = {
  id: "id",
  name: "name"
};

export const PatientObject: PatientType = {
  id: "",
  name: "",
};

export function isPatientType(object: any): object is PatientType {
  return "name" in object;
}
