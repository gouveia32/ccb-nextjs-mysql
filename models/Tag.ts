export interface TagModel {
  id: string;
  name: string;
}

export type TagType = TagModel;

export const cTagModel = {
  id: "id",
  name: "name"
};

export const TagObject: TagType = {
  id: "",
  name: "",
};

export function isTagType(object: any): object is TagType {
  return "name" in object;
}
