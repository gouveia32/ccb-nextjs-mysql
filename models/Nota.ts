import { TipoPaciente } from "./Paciente";
import { TipoControle } from "./ControleObject";
import { Nota } from "@prisma/client";

export enum tipoNotaEnum {
  CHECK = "CHECK",
  TEXT = "TEXT",
}

export interface NotaModel {
  id?: string;
  nome: string;
  tipoNota: tipoNotaEnum;
  conteudo: string;
  cor: string;
  pino: boolean;
  criadoEm: string;
  pacientes: TipoPaciente[];
  controles?: TipoControle[];
}

export type TipoNota = NotaModel;

export const NotaObject: TipoNota = {
  id: "",
  nome: "",
  tipoNota: tipoNotaEnum.TEXT,
  conteudo: "",
  cor: "#fff",
  pino: false,
  criadoEm: new Date(Date.now()).toString(),
  pacientes: [],
  controles: [],
};

export const NotaDBObject: Nota = {
  id: "",
  nome: "",
  tipoNota: "TEXT",
  cor: "",
  conteudo: "",
  pino: false,
  criadoEm: new Date(Date.now()),
  medicoId: "",
};

export enum cNotaModel {
  id = "id",
  nome = "nome",
  tipoNota = "tipoNota",
  cor = "cor",
  conteudo = "conteudo",
  pino = "pino",
  criadoEm = "criadoEm",
  pacientes = "pacientes",
  controles = "controles",
}
