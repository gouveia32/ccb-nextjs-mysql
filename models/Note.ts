import { TagType } from "./Tag";
import { CheckPointType } from "./CheckPointObject";
import { Note } from "@prisma/client";

export enum NoteTypeEnum {
  CHECK = "CHECK",
  TEXT = "TEXT",
}

export interface NoteModel {
  id?: string;
  name: string;
  noteType: NoteTypeEnum;
  content: string;
  color: string;
  pinned: boolean;
  createdAt: string;
  tags: TagType[];
  checkPoints?: CheckPointType[];
}

export type NoteType = NoteModel;

export const NoteObject: NoteType = {
  id: "",
  name: "",
  noteType: NoteTypeEnum.TEXT,
  content: "",
  color: "#fff",
  pinned: false,
  createdAt: new Date(Date.now()).toString(),
  tags: [],
  checkPoints: [],
};

export const NoteDBObject: Note = {
  id: "",
  name: "",
  noteType: "TEXT",
  color: "",
  content: "",
  pinned: false,
  createdAt: new Date(Date.now()),
  doctorId: "",
};

export enum cNoteModel {
  id = "id",
  name = "name",
  noteType = "noteType",
  color = "color",
  content = "content",
  pinned = "pinned",
  createdAt = "createdAt",
  tags = "tags",
  checkPoints = "checkPoints",
}
