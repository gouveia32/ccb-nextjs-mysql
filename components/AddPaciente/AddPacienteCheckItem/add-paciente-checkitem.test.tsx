import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import PacienteCheckItem, {
  PacienteCheckItemProps,
} from "./add-paciente-checkitem.component";
import { CheckPointObject } from "../../../models/ControleObject";
import { Checkbox, IconButton, TextField } from "@material-ui/core";

describe("AddPacienteCheckItem component", () => {
  let wrapper: ShallowWrapper;
  let mockOnDelete: any;
  let mockOnHandleChange: any;

  beforeEach(() => {
    mockOnDelete = jest.fn();
    mockOnHandleChange = jest.fn();

    const mockProps: PacienteCheckItemProps = {
      checkItem: CheckPointObject,
      onDelete: mockOnDelete,
      onHandleChange: mockOnHandleChange,
    };

    wrapper = shallow(<PacienteCheckItem {...mockProps} />);
  });

  it("should render AddPacienteCheckItem component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call onHandleChange when Checkbox component is clicked", () => {
    wrapper.find(Checkbox).simulate("change", { target: { checked: true } });

    expect(mockOnHandleChange).toHaveBeenCalled();
  });

  it("should call onHandleChange when TextField component is clicked", () => {
    wrapper.find(TextField).simulate("change", { target: { value: "" } });

    expect(mockOnHandleChange).toHaveBeenCalled();
  });

  it("should call onDelete when IconButton component is clicked", () => {
    wrapper
      .find(IconButton)
      .simulate("click", { stopPropagation: () => undefined });

    expect(mockOnDelete).toHaveBeenCalled();
  });
});
