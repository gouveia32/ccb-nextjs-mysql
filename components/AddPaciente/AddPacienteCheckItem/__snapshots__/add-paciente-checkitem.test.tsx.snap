// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddPacienteCheckItem component should render AddPacienteCheckItem component 1`] = `ShallowWrapper {}`;
